api/v1/databaseengine_types.go

```diff
@@ -39,13 +39,15 @@ type (
 
 // DatabaseEngineSpec is a spec for a database engine.
 type DatabaseEngineSpec struct {
-	Type EngineType `json:"type"`
+	Type            EngineType `json:"type"`
+	EnabledVersions Versions   `json:"enabledVersions,omitempty"`
 }
 
 // DatabaseEngineStatus defines the observed state of DatabaseEngine.
 type DatabaseEngineStatus struct {
-	State   EngineState `json:"status,omitempty"`
-	Version string      `json:"version,omitempty"`
+	State             EngineState `json:"status,omitempty"`
+	Version           string      `json:"version,omitempty"`
+	AvailableVersions Versions    `json:"availableVersions,omitempty"`
```

Comment:
For me it's a bit confusing - `Version` and `AvailableVersions`. Although a user can figure out what it suppose to mean from the `Versions` type comment, I think the comments over the fields could be still helpful.